### PR TITLE
fix: bunch of fixes

### PR DESCRIPTION
**Pull request by sakulstra**

### CODE REVIEW

File: src/libs/pool-data-provider/providers/static-pool-data-provider.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ export function StaticPoolDataProvider({
   );
 
   const activeData = isRPCActive && rpcData ? rpcData : cachedData;
-  if ((isRPCActive && rpcDataLoading) || (!isRPCActive && cachedDataLoading)) {
+  if ((isRPCActive && rpcDataLoading && !rpcData) || (!isRPCActive && cachedDataLoading)) {
```
Comment (sakulstra) [inline src/libs/pool-data-provider/providers/static-pool-data-provider.tsx:85]: without this, there's a loading render flash whenever there's a new data polling

File: src/libs/pool-data-provider/hooks/use-pool-data.tsx [hunk 2]
```diff
@@ -94,30 +88,22 @@ export function usePoolData(
...
-        errorReserves || errorUserReserves ? RECOVER_INTERVAL : POOLING_INTERVAL
-      );
-      return () => clearInterval(intervalID);
-    } else {
-      setLoadingReserves(false);
-      setLoadingUserReserves(false);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [currentAccount, lendingPoolAddressProvider, skip]);
+  usePolling(fetchReserves, POLLING_INTERVAL, skip, [skip, poolDataProviderAddress, network]);
```
Comment (sakulstra) [inline src/libs/pool-data-provider/hooks/use-pool-data.tsx:91]: splitting this up allows us to a) have different intervals b) only refetch user reserves

File: src/libs/governance-provider/hooks/use-get-proposals-rpc.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+import { useState } from 'react';
 import { Proposal, normalize, AaveGovernanceV2Interface, Network } from '@aave/protocol-js';
 import { providers } from 'ethers';
 
 import { ProposalItem } from '../types';
 import { getProposalExpiry } from '../helper';
 
 import { useStateLoading, LOADING_STATE } from '../../hooks/use-state-loading';
-import { usePooling } from '../../hooks/use-pooling';
+import { usePolling } from '../../hooks/use-polling';
```
Comment (sakulstra) [inline src/libs/governance-provider/hooks/use-get-proposals-rpc.ts:9]: fix typo

File: src/libs/hooks/use-polling.tsx [hunk 4]
```diff
@@ -14,6 +14,11 @@ export const usePooling = (
     savedCallback.current = callback;
   }, [callback]);
 
+  useEffect(() => {
```
Comment (sakulstra) [inline src/libs/hooks/use-polling.tsx:17]: usePolling didn't have a initial execution included so there was a custom implementation everywhere

**Review by aschmidt20 — APPROVED** [review]

**Pull request merged**